Add information on queue delegate hooks (#559)
vapor/docs: 4.0/docs/queues.md

// Every day at midnight
.daily().at(.midnight)
```

## Event Delegates 
The Queues package allows you to specify `JobEventDelegate` objects that will receive notifications when the worker takes action on a job. This can be used for monitoring, surfacing insights, or alerting purposes. 

> To get started, conform an object to JobEventDelegate and implement any of the methods you'd like:

Reviewer: Nit should be monospaced and tighten up the language

```suggestion
To get started, conform an object to `JobEventDelegate` and implement any required methods
```
